components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -94,6 +102,36 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
         }
       });
     } catch (error) {
+      console.error(error);
+      if (error instanceof ResponseError) {
+        switch (error.code) {
+          case ErrorCodes.USER_BLOCKED:
+            this.redirectTo(gitpodHostUrl.with({ pathname: "/blocked" }).toString());
+            return;
+          case ErrorCodes.SETUP_REQUIRED:
+            this.redirectTo(gitpodHostUrl.with({ pathname: "/first-steps" }).toString());
+            return;
+          case ErrorCodes.USER_TERMS_ACCEPTANCE_REQUIRED:
+            const thisUrl = window.location.toString();
+            this.redirectTo(gitpodHostUrl.withApi({ pathname: "/tos", search: `mode=update&returnTo=${encodeURIComponent(thisUrl)}` }).toString());
+            return;
+          case ErrorCodes.NOT_AUTHENTICATED:
+            if (!error.data) {
+              const url = gitpodHostUrl.withApi({
+                pathname: '/login',
+                search: 'returnTo=' + encodeURIComponent(window.location.toString())
+              }).toString();
+              this.redirectTo(url);
```

Comment:
Testing this didn't work (the login flow worked, but the `returnTo` did nothing).